### PR TITLE
Add CrossProject.configureAll & rename/alias/deprecate configure/configureCross

### PR DESCRIPTION
**Pull request by dwijnand**
WDYT?

### CODE REVIEW

**Comment by dwijnand** [comment]
Actually more generally it would be useful if `copy` were public, even if it need be named something like 'copyAndWatchYourStep' or 'unsafeCopy' or whatever.

**Comment by sjrd** [comment]
Dear Jenkins, test this please

@gzm0 Do we have any infrastructure to test additional methods on `CrossProject`?

> Actually more generally it would be useful if `copy` were public, even if it need be named something like 'copyAndWatchYourStep' or 'unsafeCopy' or whatever.

That really doesn't look like a good idea to me.

**Comment by sjrd** [comment]
Dear Jenkins, test this please

**Comment by dwijnand** [comment]
>>Actually more generally it would be useful if copy were public, even if it need be named something like 'copyAndWatchYourStep' or 'unsafeCopy' or whatever.

> That really doesn't look like a good idea to me.

The reason is I'd like to use the existing 'configure' method, with some `Project => Project` functions that I can get to auto-lift to `CrossProject => CrossProject`. That way I can consume my straight-forward `Project => Project` functions, but also be more JVM/JS specific where needed (eg adding `junit-interface` vs enabling `ScalaJSJUnitPlugin`).

**Comment by dwijnand** [comment]
Doh.. just realised this is already achievable in user-land with `.jvmConfigure` and `.jsConfigure`.

**Comment by japgolly** [comment]
My 2 cents: I'd like to see this in place as `.bothConfigure`. I'd like to see all the `.{jvm,js}Xxx` methods become `.{jvm,js,both}Xxx`.

**Comment by gzm0** [comment]
> Do we have any infrastructure to test additional methods on CrossProject?

Other than the `sbt-plugin-test`, I don't think so.

>  I'd like to see this in place as .bothConfigure. I'd like to see all the .{jvm,js}Xxx methods become .{jvm,js,both}Xxx.

Very valid point to have it everywhere if we have it. Suggest to replace `both` with `all`. Just in case this ever extends to another platform (native) :)

**Comment by sjrd** [comment]
Or existing convention is actually `{js,jvm,}`. See for example `settings`.

**Comment by sjrd** [comment]
To keep the conventions, we could make this an overload of `configure` with a `DummyImplicit` to fight back the double definition after erasure.

**Comment by gzm0** [comment]
> To keep the conventions, we could make this an overload of configure with a DummyImplicit to fight back the double definition after erasure.

Indeed. Maybe we should even deprecate the other one and give it a better name?

**Comment by sjrd** [comment]
Possibly, but I don't have a better name to suggest.

**Comment by dwijnand** [comment]
crossConfigure

**Comment by sjrd** [comment]
Actually, I think that `configure` is the right name for both methods, and they should meaningfully be overloaded. In both cases you configure the cross-project itself. You're just allowed to give two different types of functions as argument, but essentially the two methods are doing the same thing.

**Comment by gzm0** [comment]
> Actually, I think that configure is the right name for both methods, and they should meaningfully be overloaded.

Doesn't that mess up type-inference for argument lambdas? It'd be pretty annoying, if you had to write the type in the lambda all the time, no?

**Comment by sjrd** [comment]
Oh ... Yes, it does. Well that's not an option, then.

**Comment by sjrd** [comment]
Summarizing:
* They cannot both be called `configure`, because that messes up parameter type inference
* For consistency with other methods, ideally `configure` should be the one acting on `Project`
* The best name I've seen for the one acting on `CrossProject` would be `crossConfigure`. I would actually suggest `configureCross` as it configures the cross-project (it does not cross-configure anything).

Since we cannot remove the current `configure` for now, we cannot introduce the new `configure(Project)` now either (as, even if the old one is deprecated, it would also mess up param type inference). I therefore suggest the following plan of action:

1. In 0.6.10:
  * rename `configure(CrossProject)` into `configureCross`. Leave the old one as a deprecated alias.
  * introduce `configureAll(Project)`.
2. In 1.0.0:
  * remove the deprecated `configure(CrossProject)`
  * rename `configureAll` to `configure`, with a deprecated alias

**Comment by dwijnand** [comment]
Sounds good, and I agree to that reasoning for configureCross.

**Comment by dwijnand** [comment]
Updated.

File: sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/cross/CrossProject.scala [hunk 1]
```diff
@@ -243,9 +243,18 @@ final class CrossProject private (
   def configs(cs: Configuration*): CrossProject =
     copy(jvm.configs(cs: _*), js.configs(cs: _*))
 
-  def configure(transforms: (CrossProject => CrossProject)*): CrossProject =
+  def configureCross(transforms: (CrossProject => CrossProject)*): CrossProject =
     transforms.foldLeft(this)((p, t) => t(p))
 
+  @deprecated("Use configureCross instead.", "0.6.9")
```
Comment (sjrd) [inline sbt-plugin/src/main/scala/scala/scalajs/sbtplugin/cross/CrossProject.scala]: This should be 0.6.10 ;)

**Comment by sjrd** [comment]
That's all.

**Comment by dwijnand** [comment]
Derp. Updated.

**Comment by sjrd** [comment]
LGTM, Jenkins, test this please

**Comment by dwijnand** [comment]
nice thanks

**Pull request merged**